zcash_primitives/src/transaction/mod.rs

```diff
@@ -299,6 +302,8 @@ pub struct TransactionData<A: Authorization> {
     sprout_bundle: Option<sprout::Bundle>,
     sapling_bundle: Option<sapling::Bundle<A::SaplingAuth, ZatBalance>>,
     orchard_bundle: Option<orchard::bundle::Bundle<A::OrchardAuth, ZatBalance>>,
+    #[cfg(all(zcash_unstable = "nu7", feature = "zip-233"))]
+    zip233_amount: Zatoshis,
```

Comment:
Consider moving this to just after `expiry_height`, so the the argument order matches the order of encoding in ZIP 230.